src/util/object_factory_parameters.h

```diff
@@ -70,6 +71,14 @@ struct object_factory_parameterst
   /// Force string content to be ASCII printable characters when set to true.
   bool string_printable = false;
 
+  /// Force numerical primitive inputs to fall within the bounds
+  struct numeric_input_limitst
+  {
+    optionalt<int> lower;
+    optionalt<int> upper;
+  };
+  numeric_input_limitst assume_input_limits;
```

Comment:
I disagree - they are logically one thing (a range). However, does it make sense to have a lower but not an upper or vice versa? I'd probably make the whole thing optional if you only support having both or neither set

Reply:
I also find it more understandable if they are grouped. 

Originally, I had the whole thing as optional but I think supporting only one limit could be quite useful (allows to for example force using only positive values). Alternatively, at one point I had default values for the limits - min and max long. However, asking whether the optional has a value is I think nicer than comparing the value to max long. I also thought of making the whole thing as well as the individual limits optional but that seems like an overkill :slightly_smiling_face:

Reply:
In that case "range" or "interval" should probably appear in the name. 
Rather that defining a new type you should reuse existing ones, here `integer_intervalt` would be the right one. It will avoid having to define a new type and you can make your code simpler by using `.empty()`.

Reply:
Oh yes! That looks like exactly what I need, thank you :+1: